### PR TITLE
Iopana configurable msa

### PR DESCRIPTION
**Pull request by IoanPana**
Adding new state property to builtinidentityprovider and allow MSA to update the new property.

### CODE REVIEW

**Comment by Danielabom** [comment]
Chatted with @IoanPana, and the PR is not ready for review. He'll reach out later.

**Comment by IoanPana** [comment]
CSDL changes: https://dev.azure.com/msazure/One/_git/AD-AggregatorService-Workloads/pullrequest/6905649
Internal roadmap item: https://identitydivision.visualstudio.com/Engineering/_workitems/edit/1841904
Purpose/objective of the PR: Making MSA IdentityProvider configurable at tenant level.

**Comment by Lauragra** [comment]
@IoanPana , please only set the ready to merge label after your content reviewer has approved your PR and your APIs are publicly available. If your APIs have shipped but your PR is not yet approved, you can set the schema live label.

File: api-reference/beta/api/builtinidentityprovider-get.md [hunk 1]
```diff
@@ -0,0 +1,94 @@
+---
```
Comment (FaithOmbongi) [inline api-reference/beta/api/builtinidentityprovider-get.md:1]: Delete file as per the explanation I gave in the resource doc

File: api-reference/beta/api/builtinidentityprovider-list.md [hunk 2]
```diff
@@ -0,0 +1,96 @@
+---
```
Comment (FaithOmbongi) [inline api-reference/beta/api/builtinidentityprovider-list.md:1]: Delete file as per the explanation I gave in the resource doc

File: api-reference/beta/api/builtinidentityprovider-update.md [hunk 3]
```diff
@@ -0,0 +1,96 @@
+---
```
Comment (FaithOmbongi) [inline api-reference/beta/api/builtinidentityprovider-update.md:1]: Delete file as per the explanation I gave in the resource doc.

Instead, update the [Update identityProvider](https://learn.microsoft.com/en-us/graph/api/identityproviderbase-update?view=graph-rest-beta) doc by indicating where applicable about the only update operation allowed which is update for the MSA IdP.

File: changelog/Microsoft.CPIM.json [hunk 4]
```diff
@@ -0,0 +1,30 @@
...
+          "ChangeType": "Addition",
+          "Description": "Added the **identityProviderState** enumeration type.",
+          "Target": "identityProviderState"
+        },
+        {
+          "Id": "ea97a289-11cc-4238-8c64-66ea732ad207",
+          "ApiChange": "Property",
+          "ChangedApiName": "state",
+          "ChangeType": "Addition",
+          "Description": "Added the `state` property to the [builtInIdentityProvider](https://docs.microsoft.com/en-us/graph/api/resources/builtInIdentityProvider?view=graph-rest-beta) resource.",
```
Comment (FaithOmbongi) [inline changelog/Microsoft.CPIM.json:18]: Recommended enhancement.
```suggestion
          "Description": "Added the **state** property to the [builtInIdentityProvider](https://docs.microsoft.com/en-us/graph/api/resources/builtInIdentityProvider?view=graph-rest-beta) resource. Use this property to enable or disable the Microsoft accounts built-in identity provider in your tenant.",
```

**Review by FaithOmbongi — CHANGES_REQUESTED** [review]
Thank you @IoanPana for your work in this PR.

Some clarification: Based on our initial documentation of the APIs which we handled with Namrata, we don't need to document new APIs. Instead, all functionality for the derived types should piggy back on the existing APIs. Therefore, please delete the three new files in the api subfolder.

For update builtInIdentityProvider: Because you can only update the state property of a buildInIdentityProvider type and the property only applies to MSA IdP, make the changes to this doc instead [Update identityProvider](https://learn.microsoft.com/en-us/graph/api/identityproviderbase-update?view=graph-rest-1.0&tabs=http). Feel free to add an Example 4 to show what you can update for builtInIdentityProvider.

In addition, in [response object of Example 1 of List configured identity providers](https://learn.microsoft.com/en-us/graph/api/identitycontainer-list-identityproviders?view=graph-rest-1.0&tabs=http#response-1), add the state property to the MicrosoftAccount response object. Same sentiment for the [response object of Example 2 of Get identityProvider](https://learn.microsoft.com/en-us/graph/api/identityproviderbase-get?view=graph-rest-beta&tabs=http#example-2-retrieve-a-specific-built-in-identity-provider-only-for-azure-ad).

Please feel free to ping me internally for any clarification.

**Comment by FaithOmbongi** [comment]
The changes are live. Removing the changes requested label to avoid unactionable notifications.

**Pull request closed without merging**